src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -2414,45 +2414,40 @@ private static byte[] ConvertEnvVarsToByteArray(StringDictionary sd)
 
         private void SetStartupInfo(ProcessStartInfo startinfo, ref ProcessNativeMethods.STARTUPINFO lpStartupInfo, ref int creationFlags)
         {
+            bool hasRedirection = false;
             // RedirectionStandardInput
             if (_redirectstandardinput != null)
             {
+                hasRedirection = true;
                 startinfo.RedirectStandardInput = true;
                 _redirectstandardinput = ResolveFilePath(_redirectstandardinput);
                 lpStartupInfo.hStdInput = GetSafeFileHandleForRedirection(_redirectstandardinput, FileMode.Open);
             }
-            else
-            {
-                lpStartupInfo.hStdInput = new SafeFileHandle(ProcessNativeMethods.GetStdHandle(-10), false);
-            }
 
             // RedirectionStandardOutput
             if (_redirectstandardoutput != null)
             {
+                hasRedirection = true;
                 startinfo.RedirectStandardOutput = true;
                 _redirectstandardoutput = ResolveFilePath(_redirectstandardoutput);
                 lpStartupInfo.hStdOutput = GetSafeFileHandleForRedirection(_redirectstandardoutput, FileMode.Create);
             }
-            else
-            {
-                lpStartupInfo.hStdOutput = new SafeFileHandle(ProcessNativeMethods.GetStdHandle(-11), false);
-            }
 
             // RedirectionStandardError
             if (_redirectstandarderror != null)
             {
+                hasRedirection = true;
                 startinfo.RedirectStandardError = true;
                 _redirectstandarderror = ResolveFilePath(_redirectstandarderror);
                 lpStartupInfo.hStdError = GetSafeFileHandleForRedirection(_redirectstandarderror, FileMode.Create);
             }
-            else
+
+            if (hasRedirection)
```

Comment:
Do we really need the new variable?
```suggestion
            if (startinfo.RedirectStandardInput || startinfo.RedirectStandardOutput || startinfo.RedirectStandardError)
```

Reply:
I thought about this and believed that using the `hasRedirection` variable can avoid unneeded member access.
Essentially, the saving in overhead doesn't really matter and I just choose one style over the other.

Reply:
I noticed this because my expectations were that this variable was doing something special, but it turned out that it wasn't. Therefore, this is puzzling.